google/resource_compute_instance_group_manager.go

```diff
@@ -33,10 +33,40 @@ func resourceComputeInstanceGroupManager() *schema.Resource {
 
 			"instance_template": &schema.Schema{
 				Type:             schema.TypeString,
-				Required:         true,
+				Optional:         true,
 				DiffSuppressFunc: compareSelfLinkRelativePaths,
 			},
 
+			"version": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
```

Comment:
I think this and `instance_template` within it are also going to need to be `Computed`. When I run tests that don't set a version, it seems to want to fill one in:

```
=== RUN   TestAccInstanceGroupManager_autoHealingPolicies
--- FAIL: TestAccInstanceGroupManager_autoHealingPolicies (245.15s)
    testing.go:513: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: google_compute_instance_group_manager.igm-basic
          version.#:                   "1" => "0"
          version.0.instance_template: "https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/global/instanceTemplates/igm-test-bhvjqh6a61" => ""
        
        STATE:
        
        google_compute_http_health_check.zero:
          ID = igm-test-71h0sxbiwk
          provider = provider.google
          check_interval_sec = 1
          creation_timestamp = 2018-05-24T16:45:48.830-07:00
          description = 
          healthy_threshold = 2
          host = 
          name = igm-test-71h0sxbiwk
          port = 80
          project = hc-terraform-testing
          request_path = /
          self_link = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/global/httpHealthChecks/igm-test-71h0sxbiwk
          timeout_sec = 1
          unhealthy_threshold = 2
        google_compute_instance_group_manager.igm-basic:
          ID = igm-test-hu2dq1417r
          provider = provider.google
          auto_healing_policies.# = 1
          auto_healing_policies.0.health_check = https://www.googleapis.com/compute/beta/projects/hc-terraform-testing/global/httpHealthChecks/igm-test-71h0sxbiwk
          auto_healing_policies.0.initial_delay_sec = 10
          base_instance_name = igm-basic
          description = Terraform test instance group manager
          fingerprint = vF4q-P5d_aI=
          instance_group = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/zones/us-central1-c/instanceGroups/igm-test-hu2dq1417r
          instance_template = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/global/instanceTemplates/igm-test-bhvjqh6a61
          name = igm-test-hu2dq1417r
          named_port.# = 0
          project = hc-terraform-testing
          self_link = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/zones/us-central1-c/instanceGroupManagers/igm-test-hu2dq1417r
          target_pools.# = 1
          target_pools.1124437731 = https://www.googleapis.com/compute/beta/projects/hc-terraform-testing/regions/us-central1/targetPools/igm-test-u3zeiq2g3x
          target_size = 2
          update_strategy = RESTART
          version.# = 1
          version.0.instance_template = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/global/instanceTemplates/igm-test-bhvjqh6a61
          version.0.name = 
          version.0.target_size_fixed = 0
          version.0.target_size_percent = 0
          wait_for_instances = false
          zone = us-central1-c
        
          Dependencies:
            google_compute_http_health_check.zero
            google_compute_instance_template.igm-basic
            google_compute_target_pool.igm-basic
        google_compute_instance_template.igm-basic:
          ID = igm-test-bhvjqh6a61
          provider = provider.google
          can_ip_forward = false
          description = 
          disk.# = 1
          disk.0.auto_delete = true
          disk.0.boot = true
          disk.0.device_name = persistent-disk-0
          disk.0.disk_name = 
          disk.0.disk_size_gb = 0
          disk.0.disk_type = pd-standard
          disk.0.interface = SCSI
          disk.0.mode = READ_WRITE
          disk.0.source = 
          disk.0.source_image = debian-cloud/debian-8-jessie-v20160803
          disk.0.type = PERSISTENT
          instance_description = 
          machine_type = n1-standard-1
          metadata.% = 1
          metadata.foo = bar
          metadata_fingerprint = EEf91SmZULE=
          min_cpu_platform = 
          name = igm-test-bhvjqh6a61
          network_interface.# = 1
          network_interface.0.access_config.# = 0
          network_interface.0.address = 
          network_interface.0.alias_ip_range.# = 0
          network_interface.0.network = https://www.googleapis.com/compute/beta/projects/hc-terraform-testing/global/networks/default
          network_interface.0.network_ip = 
          network_interface.0.subnetwork = 
          network_interface.0.subnetwork_project = 
          project = hc-terraform-testing
          scheduling.# = 1
          scheduling.0.automatic_restart = true
          scheduling.0.on_host_maintenance = MIGRATE
          scheduling.0.preemptible = false
          self_link = https://www.googleapis.com/compute/beta/projects/hc-terraform-testing/global/instanceTemplates/igm-test-bhvjqh6a61
          service_account.# = 1
          service_account.0.email = default
          service_account.0.scopes.# = 3
          service_account.0.scopes.1632638332 = https://www.googleapis.com/auth/devstorage.read_only
          service_account.0.scopes.2428168921 = https://www.googleapis.com/auth/userinfo.email
          service_account.0.scopes.2862113455 = https://www.googleapis.com/auth/compute.readonly
          tags.# = 2
          tags.1996459178 = bar
          tags.2356372769 = foo
          tags_fingerprint = 
        google_compute_target_pool.igm-basic:
          ID = igm-test-u3zeiq2g3x
          provider = provider.google
          backup_pool = 
          description = Resource created for Terraform acceptance testing
          failover_ratio = 0
          health_checks.# = 0
          instances.# = 0
          name = igm-test-u3zeiq2g3x
          project = hc-terraform-testing
          region = us-central1
          self_link = https://www.googleapis.com/compute/v1/projects/hc-terraform-testing/regions/us-central1/targetPools/igm-test-u3zeiq2g3x
          session_affinity = CLIENT_IP_PROTO
FAIL
```

Reply:
I made `version` optional and computed, but  I cannot make `instance_template` Computed and Required, shouldn't we leave it Required ? (it is required within the version)

Reply:
Ah yeah, that should be fine then. I'll run the test again and check it out.

Reply:
I just ran tests and still hit the perpetual diff issue. Making version Optional & Computed is apparently not enough.

Here is the error:
```
UPDATE: google_compute_instance_group_manager.igm-basic
  version.#: "" => "<computed>"
```

Reply:
I was able to make tests pass by removing "Computed" flag, and now I see only one test failing (the one I wrote), because of the ImportStateVerify flag in my TestStep.

Do I need to add something to make version fields importable ?
```
--- FAIL: TestAccInstanceGroupManager_versions (137.02s)
        testing.go:513: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.
                
                (map[string]string) {
                }
                
                
                (map[string]string) (len=9) {
                 (string) (len=9) "version.#": (string) (len=1) "2",
                 (string) (len=27) "version.0.instance_template": (string) (len=122) "https://www.googleapis.com/compute/beta/projects/myproject/global/instanceTemplates/igm-test-q1y1jv
92jt",
                 (string) (len=14) "version.0.name": (string) (len=7) "primary",
                 (string) (len=23) "version.0.target_size.#": (string) (len=1) "0",
                 (string) (len=27) "version.1.instance_template": (string) (len=122) "https://www.googleapis.com/compute/beta/projects/myproject/global/instanceTemplates/igm-test-c4hgbmva8w",
                 (string) (len=14) "version.1.name": (string) (len=6) "canary",
                 (string) (len=23) "version.1.target_size.#": (string) (len=1) "1",
                 (string) (len=29) "version.1.target_size.0.fixed": (string) (len=1) "1",
                 (string) (len=31) "version.1.target_size.0.percent": (string) (len=1) "0"
                }               

```

Reply:
I did some debugging on my end and was able to get it working- the d.Set call for versions was failing, which is why the import test failed.

Reply:
Oh thank you, I should have catch this one, this is all my bad! Sorry for that!